homeassistant/components/velbus/config_flow.py

```diff
@@ -82,3 +85,37 @@ async def async_step_user(
             ),
             errors=self._errors,
         )
+
+    async def async_step_usb(self, discovery_info: usb.UsbServiceInfo) -> FlowResult:
+        """Handle USB Discovery."""
+        await self.async_set_unique_id(
+            f"{discovery_info.vid}:{discovery_info.pid}_{discovery_info.serial_number}_{discovery_info.manufacturer}_{discovery_info.description}"
+        )
+        dev_path = await self.hass.async_add_executor_job(
+            usb.get_serial_by_id, discovery_info.device
+        )
+        # check if this device is not already configured
+        if self._prt_in_configuration_exists(dev_path):
```

Comment:
Unrelated to this PR, but `_prt_in_configuration_exists ` could be simplified to be `return prt in velbus_entries(self.hass)`